src/components/tab-group/index.jsx

```diff
@@ -0,0 +1,73 @@
+import { useState, useId } from 'preact/hooks';
+
+import style from './style.module.css';
+
+/**
+ * Used in markdown files to switch between different API styles, i.e,
+ * Class Components and Hooks. This displays either-or, not both at once,
+ * so it's not necessarily meant for directly comparing API styles.
+ *
+ * Primarily adopted from MDN's tabs example:
+ * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#example
+ *
+ * @param {Object} props
+ * @param {string} props.tabstring - Comma-separated list of tab names
+ * @param {Array<import('preact').Component | string>} props.children - Array of JSX elements
+ */
+export default function TabGroup({ tabstring, children }) {
```

Comment:
I'm super pleased with this (almost unreasonably so), should be great for showing off different API styles/options in the future.

As noted, MDN has a brilliant tutorial in which they go over all of the accessibility complexities which I believe I mirrored correctly. Everything seems to be labeled, connected, and following practices as intended.